File: common/src/main/java/com/common/entities/enemies/Zombie.java
package com.common.entities.enemies;

import com.common.DataClasses.Position;
import com.common.entities.templates.Enemy;
import com.common.graphics.Icons;

import static com.common.Debug.Flags.Entities.Enemies.ZOMBIE;
import static com.common.Debug.Levels.CORE;
import static com.common.Debug.debug;

/**
 * Zombie entity
 */
public class Zombie extends Enemy {
	public Zombie(Position position) {
		super(position);
		debug(ZOMBIE, CORE, "--- [Zombie.constructor]");
		icon = Icons.Enemies.zombie;
		movementDelay = 2;
	}
}

File: composeApp/src/commonMain/kotlin/com/one_of_many_simons/one_button_game/entities/enemies/Zombie.kt
package com.one_of_many_simons.one_button_game.entities.enemies

import com.one_of_many_simons.one_button_game.Debug.Flags.Entities.Enemies.ZOMBIE
import com.one_of_many_simons.one_button_game.Debug.Levels.CORE
import com.one_of_many_simons.one_button_game.Debug.debug
import com.one_of_many_simons.one_button_game.dataClasses.Position
import com.one_of_many_simons.one_button_game.entities.templates.Enemy
import com.one_of_many_simons.one_button_game.graphics.Icons

/**
 * ZOMBIE entity
 */
class Zombie(position: Position) : Enemy(position) {
    init {
        debug(ZOMBIE, CORE, "--- [Zombie.constructor]")
        icon = Icons.Enemies.zombie
        movementDelay = 3
    }
}
